lib/src/posthog_widget_widget.dart

```diff
@@ -75,17 +77,32 @@ class PostHogWidgetState extends State<PostHogWidget> {
           screen: Posthog().currentScreen);
     }
 
-    // TODO: package:image/image.dart to convert to jpeg instead
+    // using png because its compressed, the native SDKs will decompress it
+    // and transform to jpeg if needed (soon webp)
+    // https://github.com/brendan-duncan/image does not have webp encoding
     final ByteData? byteData =
         await imageInfo.image.toByteData(format: ui.ImageByteFormat.png);
     if (byteData == null) {
       printIfDebug('Error: Failed to convert image to byte data.');
+      imageInfo.image.dispose();
       return;
     }
 
     Uint8List pngBytes = byteData.buffer.asUint8List();
     imageInfo.image.dispose();
 
+    if (pngBytes.isEmpty) {
+      printIfDebug('Error: Failed to convert image byte data to Uint8List.');
+      return;
+    }
+
+    if (const PHListEquality().equals(pngBytes, _lastSnapshot)) {
```

Comment:
hard to tell the performance impact of that, can move to a background thread on the native side if needed